firmware/Core/Src/mpi/mpi_command_handling.c

```diff
@@ -0,0 +1,118 @@
+#include "mpi/mpi_command_handling.h"
+#include "mpi/mpi_types.h"
+#include "debug_tools/debug_uart.h"
+#include "uart_handler/uart_handler.h"
+#include "main.h"
+#include "stm32l4xx_hal.h"
+#include "string.h"
+#include <stdio.h>
+
+enum Mpi_Mode current_mpi_mode = NOT_LISTENING_TO_MPI_MODE;
+
+/**
+ * @brief Sends commandcode+params to the MPI as bytes
+ * 
+ * @param bytes_to_send Buffer containing the CMD + Params (IF ANY) as hex bytes
+ * @param bytes_to_send_len Size of CMD buffer
+ * @param mpi_cmd_response Buffer to store incoming response from the MPI
+ * @param mpi_cmd_response_len Size of response buffer
+ * @return 0: Success, 1: Failed UART transmission, 2: Failed UART reception, 3: MPI failed to execute CMD
+ */
+uint8_t MPI_send_telecommand_hex(const uint8_t *bytes_to_send, size_t bytes_to_send_len, uint8_t mpi_response[], size_t mpi_response_size)
+{
+	// Set MPI to command mode
+	current_mpi_mode = COMMAND_MODE;
+
+	// Transmit Buffer to store CMD+Params formatted to be sent to the MPI
+	size_t UART1_txBuffer_size = bytes_to_send_len + 2;
+	//size_t UART1_rxBuffer_size = UART1_txBuffer_size + 1;
+	uint8_t UART1_txBuffer[UART1_txBuffer_size];	   	// +2 accounts for 'TC' command identifier being appended
+	memset(UART1_txBuffer, 0, UART1_txBuffer_size); 	// Initialize all elements to 0
+
+	// All commands to the MPI must begin with 'TC'
+	UART1_txBuffer[0] = 0x54;	// Hex representation of 'T'
+	UART1_txBuffer[1] = 0x43;	// Hex representation of 'C'
+
+	// Add the command bytes & parameters to the Transmit Buffer
+	if (bytes_to_send_len > 0)
+	{
+		for (size_t i = 0; i < bytes_to_send_len; i++)
+		{
+			UART1_txBuffer[2 + i] = bytes_to_send[i];
+		}
+	}
+
+	// Transmit the MPI command
+	uint32_t timeout_duration_ms = 2000;	// CONFIGURATION PARAMETER //FIXME: change to 100 later
```

Comment:
Make const. Fix FIXME. Maybe move to a global variable here.

Also, separate out RX and TX timeouts, please.